packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -110,4 +123,71 @@ export class EVMSmartWallet {
     public async nfts() {
         return this.crossmintService.fetchNFTs(this.address, this.chain);
     }
+
+    /**
+     * Sends a contract call transaction and returns the hash of a confirmed transaction.
+     * @param address the address of the contract to be called
+     * @param abi the ABI of the contract - ***should be defined as a typed variable*** to enable type checking of the contract arguments, see https://viem.sh/docs/typescript#type-inference for guidance
+     * @param functionName the name of the smart contract function to be called
+     * @param args the arguments to be passed to the function
+     * @returns The transaction hash.
+     * @throws `SendTransactionError` if the transaction fails to send. Contains the error thrown by the viem client.
+     * @throws `SendTransactionExecutionRevertedError`, a subclass of `SendTransactionError` if the transaction fails due to a contract execution error.
+     *
+     * **Passing a typed ABI:**
+     * @example
+     * const abi = [{
+     *   "inputs": [
+     *     {
+     *       "internalType": "address",
+     *         "name": "recipient",
+     *         "type": "address"
+     *       },
+     *   ],
+     *   "name": "mintNFT",
+     *   "outputs": [],
+     *   "stateMutability": "nonpayable",
+     *   "type": "function"
+     * }] as const;
+     *
+     * await wallet.sendTransaction({
+     *   address: contractAddress,
+     *   abi,
+     *   functionName: "mintNFT",
+     *   args: [recipientAddress],
+     * });
+     */
+    public async callContract<
```

Comment:
`Call` might to overloaded, it has different meanings in both
[viem](https://docs.ethers.org/v5/api/providers/provider/#Provider-call) and [ethers](https://docs.ethers.org/v5/api/providers/provider/#Provider-call)

Claude suggested `sendContractTransaction`, I think `writeContract` from `viem` also makes a lot of sense imo

Reply:
I'm mostly ambivalent but I think to the average developer callContract is clearer than writeContract at least. If you think we should change it could you write a bit more about why in the GC with Alfonso and I?